lib/BannerPlugin.js

```diff
@@ -12,6 +12,23 @@ const Template = require("./Template");
 const validateOptions = require("schema-utils");
 const schema = require("../schemas/plugins/BannerPlugin.json");
 
+/** @typedef {import("./Compiler.js")} Compiler */
+
+/**
+ * BannerPlugin Options class.
+ * @typedef {Object=} BannerOptions
```

Comment:
`{Object=}` -> `{Object}`

Reply:
When I change it to `Object` TS gives me the following:

![image](https://user-images.githubusercontent.com/966368/41669803-f1dae2f8-74b2-11e8-92f5-d359e7fb2a0e.png)

Reply:
What's this line for --> `this.options = options || {};` ?
There is a `validateOptions` call before, so I guess`undefined|null` is not valid for `options`...